features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/RumFeature.kt

```diff
@@ -438,17 +457,19 @@ internal class RumFeature(
         longTaskTrackingStrategy.unregister(appContext)
     }
 
-    private fun initializeVitalMonitors(frequency: VitalsUpdateFrequency) {
+    private fun initializeVitalMonitors(
+        frequency: VitalsUpdateFrequency
+    ): FrameStateListener? {
         if (frequency == VitalsUpdateFrequency.NEVER) {
-            return
+            return null
         }
         cpuVitalMonitor = AggregatingVitalMonitor()
         memoryVitalMonitor = AggregatingVitalMonitor()
         frameRateVitalMonitor = AggregatingVitalMonitor()
-        initializeVitalReaders(frequency.periodInMs)
+        return initializeVitalReaders(frequency.periodInMs)
     }
 
-    private fun initializeVitalReaders(periodInMs: Long) {
+    private fun initializeVitalReaders(periodInMs: Long): FrameStateListener {
```

Comment:
I'm not quite comfortable with this change, because:

* method name imposes that we initialize multiple readers, but we return only one of them
* readers we initialize not necessary related to the checking Frames state, as one may think by looking on the return type